### PR TITLE
Add ms/build01-9hdwj-worker-us-east-1a-debug

### PR DESCRIPTION
**Pull request by hongkailiu**
- How can we check if the log level is set up as expected?
- Do we need to restart crio/kubelet service on the nodes?

/cc @stevekuznetsov @mrunalp

### CODE REVIEW

File: clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+ name: debug
+spec:
+ machineConfigPoolSelector:
+   matchLabels:
+     custom-crio: log-devel-debug
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml:8]: According to [example](https://github.com/openshift/machine-config-operator/blob/master/docs/ContainerRuntimeConfigDesign.md#example),
we need to

> oc label machineconfigpool worker custom-crio=log-devel-debug

Then it applies all workers from all machinesets.
How can we apply it only to `ms/build01-9hdwj-worker-us-east-1a-debug`?

I guess the MSP will come back.

Comment (mrunalp) [reply clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml:8]: We need to create a custom pool - https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md

Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml]: Added.

Also label with 

https://github.com/openshift/release/blob/b90ae0923aada2e20f4c4c972fed8ad70d3acc18/clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml#L2-L6

File: clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  labels:
+    custom-crio: high-pid-limit
+  name: debug
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,debug]}
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml:10]: I [need to label the nodes](https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md#creating-a-custom-pool) in this ms:

> oc label node <node_name> node-role.kubernetes.io/debug=

This is the only manual step that I need to do after those objects are applied, right/

File: clusters/build-clusters/01_cluster/machine_config/kubelet-debug_machineconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    machineconfiguration.openshift.io/role: debug
+  name: kubelet-debug
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,S1VCRUxFVF9MT0dfTEVWRUw9NA==
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/machine_config/kubelet-debug_machineconfig.yaml:14]:
```bash
echo -n S1VCRUxFVF9MT0dfTEVWRUw9NA== | base64 -d
KUBELET_LOG_LEVEL=4%
```

File: clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml (hunk 2 above)
Comment (mrunalp) [reply clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml:10]: Yeah, that's what you need to do. @enxebre Can machine-api add labels to a node by default?

File: clusters/build-clusters/01_cluster/machine_config/kubelet-debug_machineconfig.yaml (hunk 3 above)
Comment (rphillips) [reply clusters/build-clusters/01_cluster/machine_config/kubelet-debug_machineconfig.yaml:14]: lgtm

**Comment by mrunalp** [comment]
@umohnani8 ptal

File: clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+  name: debug
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      custom-crio: log-devel-debug
```
Comment (umohnani8) [inline clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml:8]: possible typo: log-level instead of log-devel

**Comment by umohnani8** [comment]
Just one question, otherwise LGTM

File: clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml (hunk 4 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/machine_config/debug_ctrcfg.yaml]: Updated.

**Comment by hongkailiu** [comment]
> How are we adding a taint to these nodes?

I will do kubectl-taint after the node joins the cluster.
I do not know a way of doing this automatically.

**Comment by stevekuznetsov** [comment]
Can we check that there is no automatic way?

File: clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-debug_machine_set.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        machine.openshift.io/cluster-api-cluster: build01-9hdwj
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: build01-9hdwj-worker-us-east-1a-debug
+    spec:
+      metadata:
+        creationTimestamp: null
+        labels:
+          node-role.kubernetes.io/debug: ""
+      taints:
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-debug_machine_set.yaml:27]: https://github.com/openshift/machine-api-operator/blob/f6f04989e0f7322e56fe49e65c79e78085889b44/install/0000_30_machine-api-operator_03_machineset.crd.yaml#L370

File: clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-debug_machine_set.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: build01-9hdwj
+        machine.openshift.io/cluster-api-machine-role: worker
+        machine.openshift.io/cluster-api-machine-type: worker
+        machine.openshift.io/cluster-api-machineset: build01-9hdwj-worker-us-east-1a-debug
+    spec:
+      metadata:
```
Comment (hongkailiu) [inline clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-debug_machine_set.yaml:23]: https://github.com/openshift/machine-api-operator/blob/f6f04989e0f7322e56fe49e65c79e78085889b44/install/0000_30_machine-api-operator_03_machineset.crd.yaml#L237

**Comment by elmiko** [comment]
hey @hongkailiu and @stevekuznetsov , i just wanted to add a little info here about how to get labels and taints onto the machine and node objects.

if you set labels or annotations in MachineSet.Spec.Template.Metadata, these will get applied to every Machine object created from that MachineSet. specifically these labels and annotations will end up in Machine.Metadata.

if you set labels or annotations in the MachineSet.Spec.Template.Spec.Metadata, these will get applied to every Machine and Node object created from the MachineSet. in the case of Machine objects, these labels and annotations will be applied in Machine.Spec.Metadata. for Node objects, these values will be applied to Node.Metadata.

as for taints, if you set taints in MachineSet.Spec.Template.Spec.Taints, these will get applied to every Machine and Node object created from the MachineSet. in the case of Machine objects, the taints will appear in Machine.Spec.Taints, and for Node objects they will be on Node.Spec.Taints.

i hope this helps clear up any confusion =)

ps, i also created an addition to our FAQ with this info https://github.com/openshift/machine-api-operator/pull/582

File: clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml (hunk 2 above)
Comment (michaelgugino) [reply clusters/build-clusters/01_cluster/machine_config/debug_mcp.yaml:10]: Yes, machine-api can add labels by default.  Please refer to this section: https://github.com/openshift/machine-api-operator/blob/master/FAQ.md#adding-labels-to-nodes-via-machines

Remember, labels are only added on creation of new machines, not existing machines.

**Comment by stevekuznetsov** [comment]
@hongkailiu what is left to do here?

**Comment by hongkailiu** [comment]
Checked the comment.
I think this PR is good to go.

**Pull request merged**